src/System.CommandLine/Builder/CommandLineBuilderExtensions.cs

```diff
@@ -46,6 +46,11 @@ public static class CommandLineBuilderExtensions
         public static CommandLineBuilder AddVersionOption(
             this CommandLineBuilder builder)
         {
+            if (builder.Command.Children.GetByAlias("--version") != null)
```

Comment:
I think I'd prefer to support this as a more general capability of the `CommandLineBuilder`. 

And just to be pedantic, if the method is going to be idempotent, maybe `UseVersionOption` is a better name?